content/docs/tina-init-tutorial.md

```diff
@@ -0,0 +1,221 @@
+      path: "content/pages",
+      templates: [
+        {
+          label: "Product listing page",
+          name: "product",
+          fields: [
+            {
+              type: "group-list",
+              label: "Products",
+              name: "products",
+              fields: [
+                {
+                  type: "text",
+                  label: "Item ID",
+                  name: "id",
+                },
+                {
+                  type: "textarea",
+                  label: "Description",
+                  name: "description",
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    },
+    {
+      label: "Blog Posts",
+      name: "posts",
+      path: "content/posts",
+      templates: [
+        {
+          label: "Article",
+          name: "article",
+          fields: [
+            {
+              type: "text",
+              label: "Title",
+              name: "title",
+            },
+          ],
+        },
+      ],
+    },
+  ],
+});
+```
+
+
+Lets also create a file to query. In the folder we just created (content/pages) add a file called "product-listing.md"
+
+```md
+---
+_template: product
+---
+```
+
+
+#### Make a new Next.js page and a graphql Query
+
+Start by making a new file called pages/product-listing.tsx that contains the following
+
+```tsx
+import { LocalClient } from "tina-graphql-gateway";
+import { Pages_Document } from "../.tina/__generated__/types";
+import { AsyncReturnType } from "./demo/blog/[filename]";
+
+const ProductListingPage = (
+  props: AsyncReturnType<typeof getStaticProps>["props"]
+) => {
+  console.log(props);
+  return (
+    <div>
+      <ol>
+        {props.data.getPagesDocument?.data?.products?.map((product) => (
+          <li key={product.id}>
+            id: {product.id}, description: {product.description}
+          </li>
+        ))}
+      </ol>
+    </div>
+  );
+};
+
+export default ProductListingPage;
+
+const query = `#graphql
+query ProuctPageQuery {
+  getPagesDocument(relativePath: "content/pages/product-listing.md"){
+    data {
+      __typename ... on  Product_Doc_Data{
+        products {
+          id
+          description
+        }
+      }
+    }
+  }
+}
+`;
+const client = new LocalClient();
+
+export const getStaticProps = async () => {
+  return {
+    props: {
+      data: await client.request<{ getPagesDocument: Pages_Document }>(query, {
+        variables: {},
+      }),
+      variables: {},
+      query,
+    },
+  };
+};
+```
+
+We are doing a couple of things here. A graphql query was written that looks for the content/pages/product-listing.md file
+
+The query is being statically exported so it can be accessed by the TinaWrapper.
+
+The code was updated to display the list of products.
+
+Visit [http://localhost:3000/product-listing{" "}](http://localhost:3000/product-listing) {" "} to see what you just created. Click the "edit this site" button in the top to edit.
```

Comment:
The editor has added {" "} we should remove this.

Reply:
This is me converting JSX to markdown 😄